src/SFML/Window/CMakeLists.txt

```diff
@@ -272,11 +274,6 @@ target_link_libraries(sfml-window PUBLIC SFML::System)
 # glad sources
 target_include_directories(sfml-window SYSTEM PRIVATE "${PROJECT_SOURCE_DIR}/extlibs/headers/glad/include")
 
-# When static linking on macOS, we need to add this flag for objective C to work
-if ((NOT BUILD_SHARED_LIBS) AND SFML_OS_MACOSX)
-    target_link_libraries(sfml-window PRIVATE -ObjC)
-endif()
-
```

Comment:
If I'm understanding correctly, without us explicitly enabling OBJC and OBJCXX as languages (which was not possible when requiring 3.15), this snippet was required but once we express to CMake that we're using OBJC and OBJCXX it can automatically add this flag as required?

Reply:
The only reference I found for this flag is just to treat sources as ObjC:

https://clang.llvm.org/docs/CommandGuide/clang.html#cmdoption-objc

> Treat source input files as Objective-C and Object-C++ inputs respectively.

Reply:
The build on `master` works with this snippet removed but perhaps that's only because I'm using a version of CMake much newer than 3.15. In either case, I support its removal if it's unnecessary. I was just hoping to gain some more insight into how to best handle building Objective-C and Objective-C++ code.

Reply:
This was originally added [here](https://github.com/SFML/SFML/pull/1485) to fix static linking on macOS. With it gone, I get runtime errors when trying to use static libs. Putting it back seems to fix that problem.